src/common/types/profile.js

```diff
@@ -79,15 +156,23 @@ export type FuncTable = {
   lineNumber: Array<number|null>,
 }
 
+/**
+ * The ResourceTable holds additional information about functions. It tends to contain
+ * sparse arrays. Multiple functions can point to the same resource.
+ */
 export type ResourceTable = {
-  addonId: [any],
-  icon: [any],
+  addonId: any[], // TODO
+  icon: any[], // TODO
   length: number,
   lib: IndexIntoLibs[],
   name: IndexIntoStringTable[],
   type: resourceTypeEnum[],
 }
 
+/**
+ * Gecko has a parent process and content process(es). There can be multiple threads per
```

Comment:
I'd just say "one or more processes", because there can be lots of different kinds of processes: https://groups.google.com/d/msg/mozilla.dev.platform/6RsdfRUJz2E/i_4ymgTrAwAJ . (We currently only capture profiles for the parent, content, and gpu/compositor process, but that will hopefully change soon.)